tools/orca/src/stats.rs

```diff
@@ -0,0 +1,107 @@
+use crate::error::Error;
+use crate::run::EventRecord;
+use crossbeam::queue::{ArrayQueue, SegQueue};
+use std::sync::Arc;
+use std::thread;
+use std::time::{Duration, Instant};
+
+use mathru::statistics::distrib::{Continuous, Normal};
+
+#[derive(Debug)]
+pub enum TestPhase {
+    Start(Instant),
+    End(Instant),
+    StopNow,
+}
+
+pub trait DataCollector {
+    fn run(
+        &mut self,
+        stats_queue: Arc<SegQueue<EventRecord>>,
+        ctrl: Arc<ArrayQueue<TestPhase>>,
```

Comment:
What would the downsides be if we used a mpsc channel and just awaited on new messages instead of polling? I think this way there would be no need to do a `spawn_blocking` when calling the run function

Reply:
The tokio mpsc is quite a bit slower because it uses a semaphore to lock and submit to the queue rather than lock free submission.

Reply:
Oh cool I didn't know about that. I always thought that if you could make it asynchronous then you probably should. Like the fancier the better :p